server.go

```diff
@@ -68,22 +68,11 @@ func NewServer(config ServerConfig) (*Server, error) {
 
 	for i := range s.packetConnConfigs {
 		go func(i int, p PacketConnConfig) {
```

Comment:
I think its a little nicer to create two new functions for the whole goroutine here.
E.g.  `readPacketConn` and `readListener`.

Those functions should also get rid of the `i` variable. Thats just not a nice way to handle the `allocationManagers`. I would just to a `s.allocationManagers = append(s.allocationManagers, ...)`.